### PR TITLE
Implement dataset frame methods for per-annotation QA

### PR DESCRIPTION
**Pull request by SJLC**
Individual annotations for a dataset frame can have same Q/A operations
as the whole dataset frame: approve, request change, edit note explaining
requested change.

### CODE REVIEW

**Review by nbutler77 — CHANGES_REQUESTED** [review]
Please fix merge conflicts. Note that `add_annotations` has been implemented in `main` already (as `add_dataset_annotations`). This new functionality looks good, though it needs integration test cases. Also, please add details of the example code in `docs/usage/examples.rst`

File: examples/upload_dataset_metadata.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# example of using API to upload annotations and QA status from local copy of dataset
```
Comment (nbutler77) [inline examples/upload_dataset_metadata.py:1]: This doesn't seem like a super-useful script. What would be more useful is a script that takes a dataset id, and an arbitrary json file, and e.g. syncs the Conservator annotation QA status with the QA status in the file. Also, the name of this script is misleading, as it does not upload dataset metadata, it only makes some changes to the annotations.

**Review by nbutler77 — CHANGES_REQUESTED** [review]
Example script should probably:
a) Use new functionality added in this PR
b) Do so with a view towards demonstration, rather than being a usable script

File: examples/dataset_frame_annotation_qa_from_json.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+import sys
+
+from typing import Any, Dict, List
+
+from FLIR.conservator.conservator import Conservator
+from FLIR.conservator.wrappers.dataset_frame import DatasetFrame
+from FLIR.conservator.generated.schema import DatasetAnnotation, UpdateAnnotationInput
+
+
+# JSON format
```
Comment (shiftycow) [inline examples/dataset_frame_annotation_qa_from_json.py:15]: What's the reason for defining this JSON format?

**Review by nbutler77 — APPROVED** [review]
The example script isn't ideal, but everything else looks fine.

**Pull request merged**